extensions/wikia/ArticleAsJson/ArticleAsJson.class.php

```diff
@@ -134,13 +134,14 @@ public static function onGalleryBeforeProduceHTML( $data, &$out ){
 		return true;
 	}
 
-	public static function onPortableInfoboxNodeImageGetData( $title, &$ref, $caption ) {
-
+	public static function onPortableInfoboxNodeImageGetData( $data, &$ref ) {
 		wfProfileIn( __METHOD__ );
+
+		$title = Title::newFromText( $data['name'] );
 		if ( $title ) {
 			$details = WikiaFileHelper::getMediaDetail( $title, self::$mediaDetailConfig );
-			$details['context'] = self::MEDIA_CONTEXT_INFOBOX;
-			self::$media[] = self::createMediaObject( $details, $title->getText(), $caption );
+			$details['context'] = isset( $data['context'] ) ? $data['context'] : self::MEDIA_CONTEXT_INFOBOX;
```

Comment:
context for here image in infobox is set in infobox extension and context for regular infobox image is set here. Please consider setting both in single place.